File: ios/Sources/LocalizationSettingsPlugin/LocalizationSettingsPlugin.swift
import Foundation
import Capacitor

@objc(LocalizationSettingsPlugin)
public class LocalizationSettingsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "LocalizationSettingsPlugin"
    public let jsName = "LocalizationSettings"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getLanguage", returnType: CAPPluginReturnPromise)
        CAPPluginMethod(name: "setLanguage", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = LocalizationSettings()

    @objc func getLanguage(_ call: CAPPluginCall) {
        let value = implementation.getLanguage()
        call.resolve([
            "value": value
        ])
    }
    @objc func setLanguage(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        let result = implementation.setLanguage(value)

        call.resolve([
            "value": result
        ])
    }
}

File: android/src/main/java/com/skippay/plugins/localization/settings/LocalizationSettingsPlugin.java
package com.skippay.plugins.localization.settings;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "LocalizationSettings")
public class LocalizationSettingsPlugin extends Plugin {

    private final LocalizationSettings implementation = new LocalizationSettings();

    @PluginMethod
    public void getLanguage(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.getLanguage());
        call.resolve(ret);
    }

    @PluginMethod
    public void setLanguage(PluginCall call) {
        String value = call.getString("value");
        implementation.setLanguage(value);
        call.resolve();
    }
}
